src/content/docs/en/reference/api-reference.mdx

```diff
@@ -820,6 +820,20 @@ You can use this property to understand which route is rendering your component.
 
 See also: [`context.routePattern`](#contextroutepattern)
 
+### `Astro.isPrerendered`
+
+<p>
+
+**Type**: `string`<br />
+<Since v="5.0.0" />
+</p>
+
+A boolean representing whether or not the current page is prerendered.
+
+You can use this property to run conditional logic in middlewares, eg. to avoid accessing headers in prerendered pages.
```

Comment:
```suggestion
You can use this property to run conditional logic in middlewares, for example, to avoid accessing headers in prerendered pages.
```
I'm not sure about the language of "avoid accessing headers".... that makes it sound like accessing headers a bad thing and you want to avoid it at all costs. Is that true? Or, is it that you *can't* access headers in prerendered pages? Is this a helpful alternative for something you can't otherwise do?

Just trying to get a sense of how this is useful/helpful to know how best to phrase it!

Reply:
Not sure how to phrase this! An example: it's useless to check if a user is authenticated (eg. using a cookie which uses headers) if you generate a static og image. Also, I found out today that stuff can break if you use headers in this situation so maybe "avoid" is the right term

Reply:
OK, then I'm happy to go with this!